copy/copy.go

```diff
@@ -646,6 +689,26 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 	// If encrypted and decryption keys provided, we should try to decrypt
 	ic.diffIDsAreNeeded = ic.diffIDsAreNeeded || (isEncrypted(src) && ic.c.ociDecryptConfig != nil) || ic.c.ociEncryptConfig != nil
 
+	options.OptimizeDestinationImageAlreadyExists = true // TODO: PR: Remove forced enablement for PR testing
+	// If enabled, fetch and compare the destination's manifest. And as an optimization skip updating the destination iff equal
+	if options.OptimizeDestinationImageAlreadyExists {
+		shouldUpdateSigs := len(sigs) > 0 || options.SignBy != "" // TODO: Consider allowing signatures updates only and skipping the image's layers/manifest copy if possible
+
+		logrus.Debugf("Evaluating image manifest check optimzation: shouldUpdateSigs=%t destRequiresOciEncryption=%t ic.noPendingManifestUpdates()=%t", shouldUpdateSigs, destRequiresOciEncryption, ic.noPendingManifestUpdates())
+		if !shouldUpdateSigs && !destRequiresOciEncryption && ic.noPendingManifestUpdates() {
```

Comment:
`ic.noPendingManifestUpdates` is a bit expensive, it would be nice (and safer) to only compute it once.

Reply:
Good spot. Apologies, not enough thinking on my behalf. I've factored out.